### ISSUE TITLE
[BUG] Error cannot find 'rt/bin/RTGL1.dll on Linux

### ISSUE THREAD

**Issue by ravachol** [vs-shirokii/gzdoom-rt#10]
### Which game are you running with GZDoom?

Doom 2

### What Operating System are you using?

Linux x86_64

### Please describe your specific OS version

Manjaro

### Relevant hardware info

NVIDIA 3060

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

Can't find 'rt/bin/RTGL1.dll'.

Detailed here:

https://www.reddit.com/r/wine_gaming/comments/1ern45t/unable_to_run_doom_2_ray_traced_on_arch_through/

### Steps to reproduce the behaviour.

Explain how to reproduce
1. Start game on linux

**Comment by ravachol** [comment]
Related:
https://www.reddit.com/r/linux_gaming/comments/1er5nmp/doom_2_path_tracing_question/li38die/

**Comment by felinegirl** [comment]
from what I have found, when getting "Can't find 'rt/bin/RTGL1.dll' file", thats the rtgl dll not being able to get vulkan-1.dll

I only get this error when setting winecgf > Libaries > [add vulkan-1] > then set it to "native (windows)"
![image](https://github.com/user-attachments/assets/663466e0-19f3-4254-9f1d-a9d43537e872)

now when using anything but "native (windows)", rtgl does get loaded *in theory*, but vulkan has an issue with my gpu not having extensions
![image](https://github.com/user-attachments/assets/7200673f-1298-4d49-a541-0f3b649b990e)

the issue here is that those are windows specific extensions to vulkan on windows
all wine here is doing is passing vulkan commands to linux, but linux dosn't have the win32 extentions

ps:
people have been able to get these extensions to work running things under proton 7
[https://www.pathofexile.com/forum/view-thread/3376621](https://www.pathofexile.com/forum/view-thread/3376621)

**Comment by ravachol** [comment]
Thanks, I tried what you suggested including proton 7 but it still complains about no vulkan extensions.

**Comment by SoftwareSchlosser** [comment]
Hi, how did you manage to make it find your Steam installation of Doom?
I tried installing steam for windows in the same wine prefix but it complains about having no internet connection...

**Comment by ravachol** [comment]
I copied the wads from steam to the gzdoom-rt directory.

**Comment by KingVulpes** [comment]
Hi, would you happen to know where the config file is stored on Linux? On windows it's in Documents/My Games/GZDoom. But on Linux I can't find it

**Comment by ravachol** [comment]
Don't know.

**Comment by KingVulpes** [comment]
> Don't know.

Dang, I was asking because you can switch it to use DXGI instead of Vulkan

**Comment by ravachol** [comment]
aha

**Comment by ravachol** [comment]
/home/<user>/.config/gzdoom/gzdoom.ini. But I don't know what to change.

**Comment by KingVulpes** [comment]
> /home//.config/gzdoom/gzdoom.ini. But I don't know what to change.

Change the rt_dxgi to true

**Comment by ravachol** [comment]
hmm, can't find it, that might be a config for normal gzdoom..

**Comment by KingVulpes** [comment]
> hmm, can't find it, that might be a config for normal gzdoom..

You're right, the Doom 2 RT one is called gzdoom-rt2.ini

**Comment by KingVulpes** [comment]
> hmm, can't find it, that might be a config for normal gzdoom..

Tho maybe it's the same location? If you want I can send the Doom 2 RT one and see if putting it in the directory would work

**Comment by ravachol** [comment]
Sure.

**Comment by ravachol** [comment]
the My Documents/My Games/GZDoom/ folder is empty.

**Comment by KingVulpes** [comment]
> Sure.

[https://drive.google.com/file/d/1Rn4yfz46NUg3Qo08UEtzTPjFgUgC_eGy/view?usp=sharing](url)

**Comment by KingVulpes** [comment]
> Sure.

https://drive.google.com/file/d/1Rn4yfz46NUg3Qo08UEtzTPjFgUgC_eGy/view?usp=sharing

**Comment by ravachol** [comment]
Thank you, I will try this later.

**Comment by ravachol** [comment]
It still complains about no_vulkan_extension, unfortunately, but thanks for the help.

**Comment by KingVulpes** [comment]
> It still complains about no_vulkan_extension, unfortunately, but thanks for the help.

Darn :(

**Comment by KingVulpes** [comment]
Potential solution, put: gzdoom.exe +rt_dxgi 1 into a .bat and launch it. I can't test it since SteamOS won't even load the .exe but your distro seems to be able to load it

**Comment by ravachol** [comment]
doesn't seem to make a difference.

**Comment by SoftwareSchlosser** [comment]
> I copied the wads from steam to the gzdoom-rt directory.

I still get the error message that DOOM2.WAD couldn't be found. I copied all the wads from the "Ultimate Doom" steam game folder to the folder containing gzdoom.exe. Am I missing something else?

**Comment by ravachol** [comment]
I used the wads from doom1 + doom 2

**Comment by SoftwareSchlosser** [comment]
Yes, "Ultimate Doom" is just the name of the game folder in steam, it contins all WADs including those from doom 1 and doom 2 which I copied over to the gzdoom folder. Can't see why it doesn't work...

**Comment by ravachol** [comment]
No Idea why it doesn't work, that's what I did.

**Comment by FluffyPuppyKasey** [comment]
> Yes, "Ultimate Doom" is just the name of the game folder in steam, it contins all WADs including those from doom 1 and doom 2 which I copied over to the gzdoom folder. Can't see why it doesn't work...

All you need is Doom2.wad which is in base/doom2. Copy that ONE file right next to the EXE

**Comment by ravachol** [comment]
Maybe it's a character case issue.

**Comment by DwayneHawkins** [comment]
> Maybe it's a character case issue.

I don't think so, the casing seems OK.

**Comment by SoftwareSchlosser** [comment]
Seems related to (https://github.com/vs-shirokii/gzdoom-rt/issues/1) / (https://github.com/vs-shirokii/gzdoom-rt/issues/21).
For some reason, when installing the game in linux, it seems to download a different wad file!
After installing the game in windows and copying over the wad file from the windows partition (which is obviously the same as in AKbar30Bill's repository) the selection screen appears but after choosing Doom 2 it get's to the `cannot find 'rt/bin/RTGL1.dll` error.
Tried copying the files over from proton (Proton GE in my case -> `../files/lib64/vkd3d/libvkd3d*`) to the gzdoom-rt folder which leads to another error message box with title "Unsupported GPU" and text: "VkDevice extensiond are NOT supported: VK_KHR_external_fence_win32".
See: (https://github.com/vs-shirokii/gzdoom-rt/issues/26)

**Comment by this-is-just-con-tribute** [comment]
dang it, it's been 2 years and nothing?
